vcn_gateways.tf

```diff
@@ -79,18 +132,64 @@ resource "oci_core_route_table" "nat" {
   freeform_tags = var.tags
 
   route_rules {
+    # * With this route table, NAT Gateway is always declared as the default gateway
     destination       = local.anywhere
     destination_type  = "CIDR_BLOCK"
     network_entity_id = oci_core_nat_gateway.nat_gateway[0].id
+    description       = "Terraformed - Auto-generated at NAT Gateway creation: NAT Gateway as default gateway"
   }
 
   dynamic "route_rules" {
+    # * If Service Gateway is created with the module, automatically creates a rule to handle traffic for "all services" through Service Gateway
     for_each = var.service_gateway_enabled == true ? list(1) : []
 
     content {
       destination       = lookup(data.oci_core_services.all_oci_services[0].services[0], "cidr_block")
       destination_type  = "SERVICE_CIDR_BLOCK"
       network_entity_id = oci_core_service_gateway.service_gateway[0].id
+      description       = "Terraformed - Auto-generated at Service Gateway creation: All Services in region to Service Gateway"
+    }
+  }
+
+  dynamic "route_rules" {
+    # * filter var.nat_gateway_route_rules for routes with "drg" as destination
+    # * and steer traffic to the module created DRG
+    for_each = var.nat_gateway_route_rules != null ? { for k, v in var.nat_gateway_route_rules : k => v
+    if v.network_entity_id == "drg" } : {}
+
+    content {
+      destination       = route_rules.value.destination
+      destination_type  = route_rules.value.destination_type
+      network_entity_id = oci_core_drg.drg[0].id
+      description       = route_rules.value.description
+    }
+  }
+
+  dynamic "route_rules" {
```

Comment:
As above. I don't quite follow the purpose of this route rule. You've already filtered the NAT gw and the DRG. What is this one used for then?

Reply:
same explanation.

- 1 block to create rules on module drg if "drg" is passed,
- 1 block to create rules on module nat gateway if "nat gateway" is passed
- 1 block to create rules for any other values (gateways not managed by the module)